source/adapters/level_zero/command_buffer.cpp

```diff
@@ -241,9 +299,25 @@ ur_result_t enqueueCommandBufferMemCopyHelper(
       CommandType, CommandBuffer, NumSyncPointsInWaitList, SyncPointWaitList,
       false, RetSyncPoint, ZeEventList, ZeLaunchEvent));
 
+  // The chooseCommandList() will modify MComputeCommandListEmpty and
+  // MCopyCommandListEmpty so isFirstNode needs to be created before that.
+  bool isFirstNode{CommandBuffer->MComputeCommandListEmpty &&
+                   CommandBuffer->MCopyCommandListEmpty};
```

Comment:
rather than duplicating this variable N times across the different helpers, prefer to just have a method on the CommandBuffer class defined in the header so that it's inlined
```
 bool isFirstNode() { return MComputeCommandListEmpty && MCopyCommandListEmpty; }
```

Then at the `createSyncPointBetweenCopyAndCompute` callsites can do `isFirstNode()`

Reply:
Yeah sounds better, I'll update that.